runtime/src/snapshot_utils.rs

```diff
@@ -2900,8 +2901,26 @@ pub fn purge_old_bank_snapshots(
             })
     };
 
-    do_purge(get_bank_snapshots_pre(&bank_snapshots_dir));
-    do_purge(get_bank_snapshots_post(&bank_snapshots_dir));
+    let mut select_pre = false;
+    let mut select_post = false;
+    match filter_by_type {
+        Some(BankSnapshotType::Pre) => {
+            select_pre = true;
+        }
+        Some(BankSnapshotType::Post) => {
+            select_post = true;
+        }
+        None => {
+            select_pre = true;
+            select_post = true;
+        }
+    }
+    if select_pre {
+        do_purge(get_bank_snapshots_pre(&bank_snapshots_dir));
+    }
+    if select_post {
+        do_purge(get_bank_snapshots_post(&bank_snapshots_dir));
+    }
```

Comment:
Can do a few things here, which I think are cleaner:

1. 
```rust
    match filter_by_type {
        Some(BankSnapshotType::Pre) => do_purge(get_bank_snapshots_pre(&bank_snapshots_dir)),
        Some(BankSnapshotType::Post) => do_purge(get_bank_snapshots_post(&bank_snapshots_dir)),
        None => {
            do_purge(get_bank_snapshots_pre(&bank_snapshots_dir));
            do_purge(get_bank_snapshots_post(&bank_snapshots_dir));
        }
    }
```

2.
```rust
    if matches!(filter_by_type, Some(BankSnapshotType::Pre) | None) {
        do_purge(get_bank_snapshots_pre(&bank_snapshots_dir));
    }
    if matches!(filter_by_type, Some(BankSnapshotType::Post) | None) {
        do_purge(get_bank_snapshots_post(&bank_snapshots_dir));
    }
```

1 has repeated code, but is cleaner imo.
2 doesn't repeat, but won't yell at us if we add a new snapshot type (unlikely?)

Reply:
My earliest version was actually #1.  changed to the current one to avoid duplicated function calls.

I like your #2.

Reply:
Updated to use the recommended option #2

Reply:
another cleaning and simple way is
```
let bank_snapshots = match filter_by_type {
    Some(BankSnapshotType::Pre) => get_bank_snapshots_pre(&bank_snapshots_dir),
    Some(BankSnapshotType::Post) => get_bank_snapshots_post(&bank_snapshots_dir),
    None => get_bank_snapshots(&bank_snapshots_dir),
};
do_purge(bank_snapshots);

Reply:
> This looks beautiful to me. My preference would be this impl.

Thanks!  Updated the code to this impl.